### PR TITLE
Python 3.8 Warnings cleanup

### PR DESCRIPTION
**Pull request by franchuterivera**
Deprecated utilities:
sklearn.utils.testing is deprecated  --> numpy assert equivalent
sklearn.externals.joblib  -->use from joblib directly
sklearn.linear_model.stochastic_gradient --> from sklearn.linear_model directly
Setting a random_state has no effect since shuffle is False... --> only add random state is no shuffle
DeprecationWarning: Please use assertRaisesRegex instead. --> Removed the "p" as it is a deprecated alias  https://docs.python.org/3.2/library/unittest.html
"DeprecationWarning: invalid escape sequence \( ---> re.compile('\(')  --> Added another backslash re.compile('\\(')

### CODE REVIEW

File: test/test_data/test_data_manager.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import unittest
 
 import numpy as np
-from sklearn.utils.testing import assert_array_almost_equal
+from numpy.testing import assert_array_almost_equal
```
Comment (mfeurer) [inline test/test_data/test_data_manager.py:5]: Could you please use `np.testing`?

File: test/test_metalearning/pyMetaLearn/metalearning/test_kND.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import unittest
 import numpy as np
+from numpy.testing import assert_array_almost_equal
```
Comment (mfeurer) [inline test/test_metalearning/pyMetaLearn/metalearning/test_kND.py:3]: Same here.

File: test/test_pipeline/components/data_preprocessing/test_data_preprocessing_numerical.py [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import unittest
 import numpy as np
-from scipy import sparse
+from numpy.testing import assert_array_almost_equal
```
Comment (mfeurer) [inline test/test_pipeline/components/data_preprocessing/test_data_preprocessing_numerical.py:3]: Same here

File: test/test_pipeline/components/data_preprocessing/test_minority_coalescence.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import unittest
 import numpy as np
+from numpy.testing import assert_array_almost_equal
```
Comment (mfeurer) [inline test/test_pipeline/components/data_preprocessing/test_minority_coalescence.py:3]: Same here.

File: test/test_pipeline/implementations/test_MinorityCoalescer.py [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import unittest
 import numpy as np
+from numpy.testing import assert_array_almost_equal
```
Comment (mfeurer) [inline test/test_pipeline/implementations/test_MinorityCoalescer.py:3]: Same here.

File: test/test_pipeline/implementations/test_SparseOneHotEncoder.py [hunk 6]
```diff
@@ -1,6 +1,8 @@
 import unittest
 
 import numpy as np
+from numpy.testing import assert_array_almost_equal
```
Comment (mfeurer) [inline test/test_pipeline/implementations/test_SparseOneHotEncoder.py:4]: Same here.

File: test/test_pipeline/implementations/test_util.py [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import unittest
 
 import numpy as np
-from sklearn.utils.testing import assert_array_almost_equal
+from numpy.testing import assert_array_almost_equal
```
Comment (mfeurer) [inline test/test_pipeline/implementations/test_util.py:4]: Same here.

File: test/test_pipeline/test_classification.py [hunk 8]
```diff
@@ -6,14 +6,15 @@
 import unittest
 import unittest.mock
 
+from joblib import Memory
 import numpy as np
+from numpy.testing import assert_array_almost_equal
```
Comment (mfeurer) [inline test/test_pipeline/test_classification.py:11]: Same here.

File: test/test_pipeline/test_regression.py [hunk 9]
```diff
@@ -6,11 +6,12 @@
 import unittest.mock
 
 import numpy as np
+from numpy.testing import assert_array_almost_equal
```
Comment (mfeurer) [inline test/test_pipeline/test_regression.py:9]: Same here.
